R/separate_longer.R

```diff
@@ -0,0 +1,255 @@
+
+  # If no string columns, return data unchanged (like tidyr)
+  if (length(col_names) == 0) {
+    return(add_tidypolars_class(data))
+  }
+
+  # Split each column by delimiter and convert to list
+  out <- data$with_columns(
+    do.call(pl$col, as.list(col_names))$str$split(delim)
+  )
+
+  # Handle multi-column broadcasting and validation
+  if (length(col_names) > 1) {
+    # Convert lists containing only empty strings [""] to empty lists []
+    # This matches tidyr behavior where "" split by "," gives character(0)
+    # Only do this for multi-column case to enable proper broadcasting
+    empty_list_exprs <- lapply(col_names, function(nm) {
+      col <- pl$col(nm)
+      # Check if list has length 1 and that element is ""
+      is_empty_string <- col$list$len()$eq(1)$and(col$list$first()$eq(""))
+      # If so, replace with empty list, otherwise keep original
+      pl$when(is_empty_string)$then(
+        pl$lit(list(character(0)))$cast(pl$List(pl$String))
+      )$otherwise(col)$alias(nm)
+    })
+    out <- out$with_columns(!!!empty_list_exprs)
+
+    out <- handle_multi_column_explode(out, col_names)
+  }
+
+  # Explode all columns together
+  add_tidypolars_class(out$explode(col_names))
+}
+
+
+#' @rdname separate_longer
+#' @export
+separate_longer_position_polars <- function(
+  data,
+  cols,
+  width,
+  ...,
+  keep_empty = FALSE
+) {
+  check_polars_data(data)
+  rlang::check_dots_empty()
+  rlang::check_required(cols)
+  check_number_whole(width, min = 1)
+  check_bool(keep_empty)
+
+  # Capture the column expression and filter to only string columns
+  col_names <- tidyselect_named_arg(data, rlang::enquo(cols)) |>
+    intersect(names(data$select(pl$col(pl$String))))
+
+  # If no string columns, return data unchanged (like tidyr)
+  if (length(col_names) == 0) {
+    return(add_tidypolars_class(data))
+  }
+
+  # Use regex to extract groups of `width` characters
+  # Pattern: .{1,width} matches 1 to width characters (greedy)
+  pattern <- paste0(".{1,", width, "}")
+  out <- data$with_columns(
+    do.call(pl$col, as.list(col_names))$str$extract_all(pattern)
+  )
+
+  # Handle keep_empty
+  if (!keep_empty) {
+    cols <- do.call(pl$col, as.list(col_names))
+    condition <- cols$is_null()$or(cols$list$len()$gt(0))
+    out <- out$filter(pl$all_horizontal(condition))
+  }
+
+  # Handle multi-column broadcasting and validation
+  if (length(col_names) > 1) {
+    out <- handle_multi_column_explode(out, col_names)
+  }
+
+  # Explode all columns together
+  add_tidypolars_class(out$explode(col_names))
+}
+
+#' Helper function to handle multi-column explode with broadcasting
+#'
+#' Handles broadcasting and validation for multi-column explode operations,
+#' following tidyr behavior:
+#' - Columns with length 0 (empty) or 1 are broadcast to match the row's max length
+#' - Null values are repeated to match the max length
+#' - Empty lists (from empty strings) become [""] when max_len is 0
```

Comment:
I have a warning from `roxygen2` when re-documenting the package because of the `[""]` here. Can you fix it?